frontend/src/lib/components/UnitPicker/UnitPicker.tsx

```diff
@@ -27,125 +23,126 @@ const aggregationDisplayMap = aggregationAxisFormatSelectOptions.reduce((acc, op
                 type={'secondary'}
                 status="stealth"
                 data-attr="chart-aggregation-axis-format"
                 popup={{
-                    onClickOutside: close,
+                    onClickOutside: () => setIsVisible(false),
+                    additionalRefs: [customUnitModalRef],
                     visible: isVisible,
                     className: 'UnitPopup',
                     overlay: (
                         <>
                             {aggregationAxisFormatSelectOptions.map(({ value, label }, index) => (
                                 <LemonButton
                                     key={index}
-                                    onClick={() => handleChange({ format: value, close: true })}
+                                    onClick={() => handleChange({ format: value })}
                                     status="stealth"
                                     active={value === localAxisFormat}
                                     fullWidth
                                 >
                                     {label}
                                 </LemonButton>
                             ))}
-                            {!!featureFlags[FEATURE_FLAGS.CURRENCY_UNITS] && (
-                                <>
-                                    <LemonDivider />
-                                    <PureField label={'prefix:'}>
-                                        <LemonInput
-                                            value={localAxisPrefix}
-                                            onChange={(prefix) => handleChange({ prefix })}
-                                            onPressEnter={(prefix) => handleChange({ prefix, close: true })}
-                                        />
-                                    </PureField>
-                                    <PureField label={'postfix:'}>
-                                        <LemonInput
-                                            value={localAxisPostfix}
-                                            onChange={(postfix) => handleChange({ postfix })}
-                                            onPressEnter={(postfix) => handleChange({ postfix, close: true })}
-                                        />
-                                    </PureField>
-                                </>
-                            )}
+
+                            <>
+                                <LemonDivider />
+                                <LemonButton
+                                    onClick={() => setCustomUnitModal('prefix')}
+                                    status="stealth"
+                                    active={!!filters.aggregation_axis_prefix}
+                                    fullWidth
+                                >
+                                    Custom prefix...
+                                    {!!filters.aggregation_axis_prefix ? `: ${filters.aggregation_axis_prefix}` : ''}
```

Comment:
The ellipsis needs to be after the current value
<img width="314" alt="Screenshot 2022-11-02 at 12 49 47" src="https://user-images.githubusercontent.com/4550621/199482679-4c4235b7-92ad-4d0d-a9b1-266d9801e69b.png">